services/uploads.js

```diff
@@ -113,19 +113,29 @@ Uploads.createUploadCredentials = function() {
  */
 Uploads.createUpload = function(config) {
   v.assertShape({
-    mapId: v.required(v.string),
+    tileset: v.required(v.string),
     url: v.required(v.string),
-    tilesetName: v.string
+    name: v.string,
+    private: v.boolean
   })(config);
 
+  var body = {
+    tileset: config.tileset,
+    url: config.url
+  };
+
+  if (config.name) body.name = config.name;
+
+  if (config.private === false) {
+    body.private = false;
+  } else {
+    body.private = true;
+  }
+
   return this.client.createRequest({
     method: 'POST',
     path: '/uploads/v1/:ownerId',
-    body: {
-      tileset: config.mapId,
-      url: config.url,
-      name: config.tilesetName
-    }
+    body: body
```

Comment:
To keep the codebase consistent could you use pick to set the body? See https://github.com/mapbox/mapbox-sdk-js/blob/fc74a74f6c48d7373aec06548379f0c534f4d298/services/tilesets.js#L40-L58 for an example of how it's done throughout the rest of the codebase.

Reply:
[This comment](https://github.com/mapbox/mapbox-sdk-js/pull/373#discussion_r430472960) speaks on using `pick` a bit. I think more work would be required to use `pick`.